API/routers/publish_router.py

```diff
@@ -63,7 +63,7 @@ async def post_publishers(self,
                                   default=Required,
                                   description="payload_category is like a topic, table or bucket or directory name."
                                               "Used when writing data to the destination",
-                                  regex=r"^[a-zA-Z][a-zA-Z0-9_]*$"
+                                  regex=r"^[a-zA-Z][a-zA-Z0-9_-]*$"
```

Comment:
BigQuery, Presto and some databases doesn't allow hyphen (-) in the table names, so it is better remove hyphen in the regex